examples/plugins/BUILD.bazel

```diff
@@ -0,0 +1,80 @@
+load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_files")
+load("@aspect_bazel_lib//lib:directory_path.bzl", "directory_path")
+load("@aspect_bazel_lib//lib:copy_file.bzl", "copy_file")
+load("@aspect_rules_swc//swc:defs.bzl", "swc", "swc_plugin")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+
+npm_link_all_packages(name = "node_modules")
+
+directory_path(
```

Comment:
this file needs more comments to serve as an example. There are two different approaches here, right? Most users would use a pattern like `npm_plugin`, while the `file_plugin` would be for advanced users who know about building web assembly or want to code in Rust.